rtos/Thread.cpp

```diff
@@ -265,11 +268,7 @@ uint32_t Thread::max_stack() {
     _mutex.lock();
 
     if (_tid != NULL) {
-        os_thread_t *thread = (os_thread_t *)_tid;
-        uint32_t high_mark = 0;
-        while (((uint32_t *)(thread->stack_mem))[high_mark] == 0xE25A2EA5)
-            high_mark++;
-        size = thread->stack_size - (high_mark * sizeof(uint32_t));
+        size = osThreadGetStackSize(_tid) - osThreadGetStackSpace(_tid);
```

Comment:
to use osGetStackSpace we need to enable OS_STACK_WATERMARK, right now we only do it conditionally.